src/software/SfM/main_SfMInit_ImageListing.cpp

```diff
@@ -427,13 +436,13 @@ int main(int argc, char **argv)
       sCamModel = exifReader.getModel();
       focalLength_mm = exifReader.getFocal();
       // Handle case where focal length is equal to 0
-      if (focalLength_mm == 0.0f)
+      if (focalLength_mm <= 0.0f)
       {
         error_report_stream
           << stlplus::basename_part(imageAbsFilepath) << ": Focal length is missing in metadata." << "\n";
         focalPix = -1.0;
       }
-      else
+      else if(ccdw != 0.0)
       {
         // Retrieve the focal from the metadata in mm and convert to pixel.
         focalPix = std::max ( metadataImageWidth, metadataImageHeight ) * focalLength_mm / ccdw;
```

Comment:
maybe add an `else` to write a warning message to manage the error case (at least for debugging purpose, it could be that for some reason the focal > 0 but we don't have the info about the ccd size)